src/node.rs

```diff
@@ -238,25 +237,24 @@ pub struct Element {
 
 impl Element {
     #[doc(hidden)]
-    pub fn new(name: QualName, attrs: Vec<Attribute>) -> Self {
-        let id = attrs
-            .iter()
-            .find(|a| a.name.local.deref() == "id")
-            .map(|a| LocalName::from(a.value.deref()));
-
-        let classes: HashSet<LocalName> = attrs
-            .iter()
-            .find(|a| a.name.local.deref() == "class")
-            .map_or_else(HashSet::new, |a| {
-                a.value
-                    .deref()
-                    .split_whitespace()
-                    .map(LocalName::from)
-                    .collect()
-            });
+    pub fn new(name: QualName, attributes: Vec<Attribute>) -> Self {
+        let mut classes: HashSet<LocalName> = HashSet::new();
+        let mut id: Option<LocalName> = Option::default();
+        let mut attrs = Attributes::with_capacity(attributes.len());
+
+        for a in attributes {
+            if a.name.local.deref() == "id" {
```

Comment:
Maybe `match a.name.local.deref() { .. }` to avoid the repetition?

Reply:
@adamreichold before I saw this, actually updated it to deref once to avoid the penalties.

Reply:
The `.deref()` call is basically free, from my PoV this is about readability. Hence, I also find the `value` dance too much as it obfuscates the code.

Reply:
> Hence, I also find the value dance too much as it obfuscates the code.

Ok, that seems to be gone anyway. Sorry for the noise.

Reply:
reverted for readability since its minor for this